custom/icds/translations/integrations/transifex.py

```diff
@@ -83,3 +93,64 @@ def _cleanup(self):
         for resource_name, filepath in self.po_file_generator.generated_files:
             if os.path.exists(filepath):
                 os.remove(filepath)
+
+    def _get_resource_slugs_for_version(self, version):
+        """
+        :param version: version number
+        :return: list of resource slugs corresponding to version
+        """
+        return [r['name']
+                for r in self.client().list_resources().json()
+                if r['name'].endswith("v%s" % version)]
+
+    @staticmethod
+    def _ensure_resource_slugs_for_version(resource_slugs, version):
+        """
+        confirms that resource slugs provided are for the expected version by checking for its name to end with
+        v[version number] like v15 for version 15.
+        :param resource_slugs: list of resource slugs
+        :param version: version
+        """
+        for resource_slug in resource_slugs:
+            if not resource_slug.endswith("v%s" % version):
+                raise Exception("Resource name '{}' is expected to contain version".format(
+                    resource_slug
+                ))
+
+    def get_translations(self, resource_slugs=None):
+        """
+        pull translations from transifex
+        :param resource_slugs: optional argument. All resource slugs corresponding to version are used otherwise.
+        :return: list of POEntry objects
+        """
+        if resource_slugs:
+            self._ensure_resource_slugs_for_version(resource_slugs, self.version)
```

Comment:
I think this check should go in the code that fetches the `resource_slugs` - one place that's responsible for returning valid slugs, so everywhere else knows what they're working with.

Reply:
It is called with resource_slugs as `None`. So if there are no resource slugs it tries to find them first and if its still missing, then it raises an `Exception`.

Reply:
`resource_slugs` can be passed to this function as an argument. Currently there is no place where its fetched and then passed to this method.